### PR TITLE
fix empty logging and cgroup

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rancher/issues/6277

**Comment by StrongMonkey** [comment]
@alena1108 @joshwget @deniseschannon fix three bugs:

- remove `logging` when no driver name is specified

- remove `cgroup`

- remove `:` when no options is passed to `tmpfs`

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rancher/issues/6461

**Comment by cjellick** [comment]
`cgroup` isn't a docker field. Remove it from the cattle API.

File: code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/service/impl/ServiceDiscoveryApiServiceImpl.java [hunk 1]
```diff
@@ -322,7 +324,12 @@ private void populateTmpfs(Map<String, Object> cattleServiceData, Map<String, Ob
                     if (map.get(key) != null) {
                         option = map.get(key).toString();
                     }
-                    list.add(key + ":" + option);
+                    if (option.isEmpty()) {
+                        list.add(key);
+                    }
+                    else {
```
Comment (cjellick) [inline code/iaas/service-discovery/api/src/main/java/io/cattle/platform/servicediscovery/api/service/impl/ServiceDiscoveryApiServiceImpl.java]: Stylistically, if elses should look like:
```
if (condition) {
    // something
} else {
    // something else
}
```
meaning: put the else statement on the same line as the closing bracket for the if

File: tests/integration/cattletest/core/test_svc_discovery.py [hunk 2]
```diff
@@ -2133,7 +2131,11 @@ def test_malform_export_config(client, context):
                              'readIops': 1000,
                              'writeIops': 2000,
                          },
-                     }}
+                     },
+                     "cgroup": "mygroup",
```
Comment (cjellick) [inline tests/integration/cattletest/core/test_svc_discovery.py]: You're going to remove cgroup from the cattle API, so I assume this will go away.

**Comment by cjellick** [comment]
LGTM

**Pull request merged**